File: src/main/java/org/redrune/engine/tick/task/WorldTaskInformation.java
package org.redrune.engine.tick.task;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/6/2017
 */
final class WorldTaskInformation {
	
	/**
	 * The task to run
	 */
	private WorldTask task;
	
	/**
	 * The initial delay, in ticks, that the task waits before starting.
	 */
	private int initialTickDelay;
	
	/**
	 * The delay, in ticks, that the task waits after it has ran once.
	 */
	private int repeatTickDelay;
	
	public WorldTaskInformation(WorldTask task, int initialTickDelay, int repeatTickDelay) {
		this.task = task;
		this.initialTickDelay = initialTickDelay;
		this.repeatTickDelay = repeatTickDelay;
		if (repeatTickDelay == -1) {
			task.needRemove = true;
		}
	}

    public WorldTask getTask() {
        return this.task;
    }

    public int getInitialTickDelay() {
        return this.initialTickDelay;
    }

    public int getRepeatTickDelay() {
        return this.repeatTickDelay;
    }

    public void setTask(WorldTask task) {
        this.task = task;
    }

    public void setInitialTickDelay(int initialTickDelay) {
        this.initialTickDelay = initialTickDelay;
    }

    public void setRepeatTickDelay(int repeatTickDelay) {
        this.repeatTickDelay = repeatTickDelay;
    }
}

File: src/main/kotlin/org/redrune/engine/tick/task/WorldTaskInformation.kt
package org.redrune.engine.tick.task

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/6/2017
 */
internal class WorldTaskInformation(
    /**
     * The task to run
     */
    var task: WorldTask,
    /**
     * The initial delay, in ticks, that the task waits before starting.
     */
    var initialTickDelay: Int,
    /**
     * The delay, in ticks, that the task waits after it has ran once.
     */
    var repeatTickDelay: Int,
) {

    init {
        if (repeatTickDelay == -1) {
            task.isNeedRemove = true
        }
    }
}